packages/react-checkbox/Spec.md

```diff
@@ -2,62 +2,267 @@
+|styles|Additional styles.|
+|toggle|Render a toggle style checkbox with on and off choices.|
+|variables|Additional styles.|
+
+### [Checkbox in v8/Fabric](https://developer.microsoft.com/en-us/fluentui#/controls/web/checkbox)
+
+The v8 `Checkbox` component supports both `indeterminate` and `checked` states. In this case, an input tag is used and its opacity is set to 0. This allows for the logic to be performed by the native element and a div is rendered to show the styled checkbox.
+
+Example
+
+```tsx
+<Checkbox
+  label="Indeterminate checkbox (controlled)"
+  indeterminate={isIndeterminate}
+  checked={isChecked}
+  onChange={onChange}
+/>
+```
+
+Component props:
+|Prop|Description|
+|---|---|
+|ariaDescribedBy|Id of the element that describes the checkbox.|
+|ariaLabel|Accessible label for the checkbox.|
+|ariaLabelledBy|Id of the element that contains the label information of the checkbox.|
+|ariaPositionInSet|Number in the parent set (if in a set) for aria-posinset.|
+|ariaSetSize|The total size of the parent set (if in a set) for aria-setsize.|
+|boxSide|Wheter the checkbox should be shown before or after the label.|
+|checked|Checkbox's checked state.|
+|checkmarkIconProps|Icon to be rendered in the checkbox.|
+|className|Additional styles.|
+|componentRef|Optional ref.|
+|defaultChecked|Whether the checkbox should be set to checked by default.|
+|defaultIndeterminate|Whether the checkbox should be set to indeterminate by default.|
+|disabled|Whether the checkbox is disabled or not.|
+|id|Id for the checkbox input.|
+|indeterminate|Checkbox's indeterminate state.|
+|inputProps|Optional props to be applied to the input element before applying other props.|
+|label|String to display next to the checkbox.|
+|name|Name for the checkbox input.|
+|onChange|Event handler to be called after the checked value has changed.|
+|onRenderLabel|Custom render for the label.|
+|required|Required state of the checkbox.|
+|styles|Additional styles.|
+|theme|Additional styles.|
+|title|Title text applied to the root element and the hidden checkbox input.|
+
+### Conclusion
+
+- Most props will follow the approach of `v8` as well as the render structure. The main idea will be to use the native `input` tag and set its opacity to 0 to then render a custom checkbox.
+- This approach will also use the `Label` component from `@fluentui/react-label`
+- The converged `Checkbox` should also support `circular` checkboxes which both `v8` and `v0` do not support.
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+```tsx
+<Checkbox label="Example Checkbox"/>
+
+<Checkbox label={<>Example Checkbox with <a href="https://www.microsoft.com">link</a></>} />
+
+<Checkbox circular size="large" label="Circular Checkbox" />
+
+<Checkbox label="Controlled Checkbox" onChange={onChangeFunction} />
+
+<Checkbox indeterminate label="Indeterminate Checkbox" />
```

Comment:
```suggestion
<Checkbox checked='indeterminate' label="Indeterminate Checkbox" />
```

According to the spec it should be `checked='indeterminate'`, correct?

Reply:
this is an interesting one, since the native `input` checkbox supports both `checked` and `indeterminate` attributes although `indeterminate` is an IDL attribute. I would rather be in favour of following the native `input` as much as possible